Orchestrator build fails on macOS presumably due to linux function call dependency

**Description**
Following the official [[build instructions](https://apps.scion.org/)](https://apps.scion.org/) for macOS, the `scion-orchestrator` build fails when running `go build`.

**Steps to Reproduce**

1. Clone repositories and checkout version:

   ```bash
   git clone https://github.com/netsys-lab/scion-pila
   git clone https://github.com/netsys-lab/scion-orchestrator && cd scion-orchestrator
   mkdir bin && mkdir dev && cd dev
   git clone https://github.com/scionproto/scion.git
   cd scion
   git checkout v0.12.0
   cd ../../
   ```

2. Build orchestrator:

   ```bash
   go build
   ```

**Error log**

```text
pkg/apiv1/services_handler.go:27:25: undefined: osutils.GetJournalLogs
pkg/apiv1/services_handler.go:43:25: undefined: osutils.GetJournalLogs
pkg/apiv1/services_handler.go:59:25: undefined: osutils.GetJournalLogs
pkg/apiv1/services_handler.go:75:25: undefined: osutils.GetJournalLogs
pkg/apiv1/services_handler.go:92:25: undefined: osutils.GetJournalLogs
```

**Analysis**

* The error appears to be caused by references to **Linux-specific implementation** of `osutils.GetJournalLogs`.
* These calls are not available on macOS, causing the build to fail.

**Environment**

* OS: macOS 15.5 (24F74)
* Go: v1.25.0
* SCION orchestrator: v0.12.0

--- a/.gitignore
+++ b/.gitignore
@@ -42,3 +42,4 @@ bin_latest/
 config_latest/
 
 bin_release
+.DS_Store

--- a/pkg/osutils/service_log_darwin.go
+++ b/pkg/osutils/service_log_darwin.go
@@ -1,0 +1,65 @@
+package osutils
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+// GetJournalLogs fetches logs for a service between start and end positions
+func GetJournalLogs(service string, lines int) (string, error) {
+	const lastWindow = "10m"
+
+	predicate := fmt.Sprintf(
+		`process == "%[1]s" OR subsystem == "%[1]s" OR senderImagePath CONTAINS[c] "%[1]s"`,
+		service,
+	)
+
+	cmd := exec.Command(
+		"log", "show",
+		"--style", "syslog",
+		"--predicate", predicate,
+		"--last", lastWindow,
+	)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Run(); err != nil {
+		return "", err
+	}
+
+	return tailLastN(out.Bytes(), lines), nil
+}
+
+// tailLastN returns the last n lines of a byte slice
+func tailLastN(b []byte, n int) string {
+	if n <= 0 || len(b) == 0 {
+		return ""
+	}
+	sc := bufio.NewScanner(bytes.NewReader(b))
+	const maxLine = 1024 * 1024
+	buf := make([]byte, 0, 64*1024)
+	sc.Buffer(buf, maxLine)
+
+	ring := make([]string, n)
+	idx, count := 0, 0
+	for sc.Scan() {
+		ring[idx] = sc.Text()
+		idx = (idx + 1) % n
+		if count < n {
+			count++
+		}
+	}
+
+	var res bytes.Buffer
+	start := (idx + (n - count)) % n
+	for i := 0; i < count; i++ {
+		pos := (start + i) % n
+		res.WriteString(ring[pos])
+		res.WriteByte('\n')
+	}
+	return res.String()
+}

